### PR TITLE
feat: update coinbase handling for new tx and output validation

### PR DESCRIPTION
**Pull request by philipr-za**
Description
---
The new tx validation protocol would continuously try and revalidate abandoned coinbase transactions and the TXO validation would try validate abandoned or reforged Coinbases indefinitely.

This PR updates the Tx validation handing of Coinbases to give them the Coinbase status when they are abandoned which means they won’t be revalidated but will still be checked during a reorg. The TXO validation has been updated so that ab abandoned coinbase can be marked as Abandoned when the transaction validation detects the abandonment so that the output will not be validated indefinitely.

A new test is provided and a number of cucumber tests for this branch have been fixed.

How Has This Been Tested?
---
Unit test provided and cucumber tests updated

### CODE REVIEW

**Review by SWvheerden — APPROVED** [review]
LGTM

File: base_layer/wallet/src/output_manager_service/storage/models.rs [hunk 1]
```diff
@@ -131,4 +131,5 @@ pub enum OutputStatus {
     ShortTermEncumberedToBeReceived,
     ShortTermEncumberedToBeSpent,
     SpentMinedUnconfirmed,
+    AbondonedCoinbase,
```
Comment (stringhandler) [inline base_layer/wallet/src/output_manager_service/storage/models.rs]:
```suggestion
    AbandonedCoinbase,
```

Comment (philipr-za) [reply base_layer/wallet/src/output_manager_service/storage/models.rs]: oof, nice catch

Comment (stringhandler) [reply base_layer/wallet/src/output_manager_service/storage/models.rs]: Happy with it otherwise

**Review by stringhandler — APPROVED** [review]

**Pull request merged**